impl/socket.cxx

```diff
@@ -22,7 +22,7 @@
 using substrate::INVALID_SOCKET;
 inline int closesocket(const int s) { return close(s); }
 #else
-#	include <Winsock2.h>
+#	include <WinSock2.h>
```

Comment:
Both the original and the new name are not correct. This makes the inclusion fail if you ever use a case sensitive filesystem. For both [the MinGW SDK](https://github.com/mingw-w64/mingw-w64/blob/master/mingw-w64-headers/include/winsock2.h) and [the Microsoft SDK](https://learn.microsoft.com/es-es/windows/win32/api/winsock2/nf-winsock2-socket), the inclusion should be:

```suggestion
#	include <winsock2.h>
```

Reply:
We based this change on the headers as delivered by xwin - which has `WinSock2.h` as the real file extracted from the SDK archives, and `winsock2.h` as a symlink for that name.
Which then would be better in your opinion to do, given we got bitten by case sensitivity to make this change?

Reply:
Always use the lowercase version -- it'll be mapped by Windows if needed.

Reply:
This has been fixed. Iff this winds up causing problems on something spicy like a Windows machine that's had filesystem case sensitivity enabled, we'll introduce some `__has_include()` logic to try and get the right name for the header 🤞🏼